src/locationSource/GpsApiLocationSource.js

```diff
@@ -0,0 +1,18 @@
+export default class GpsApiLocationSource {
+    async getLocation() {
+        let {coords} = await this._getCurrentPosition();
+        let {latitude, longitude} = coords;
+
+        return {latitude, longitude};
+    }
+
+    _getCurrentPosition() {
+        const expr = (resolve, reject) => {
+            navigator.geolocation ? 
+                navigator.geolocation.getCurrentPosition(resolve, reject) : 
+                resolve({});
```

Comment:
Czy tu nie powinno być jakiegoś "reject" albo "throw"?

Popraw mnie, jeśli się mylę, ale powiedzmy, że navigator.geolocation jest undefined (coś poszło nie tak).
Wtedy:
 - exppression resolvuje z pustym obiektem
 - _getCurrentPosition resolvuje (zamiast rzucić błędem) i zwraca pusty obiekt

w rezultacie (na ile ogarniam i poprawnie przetestowałem tę składnię, której używasz operując na obiektach - mogę się mylić, bo jak mam być szczery, to nie jest bardzo czytelna, przynajmniej dla mnie ;) ) getLocation() wywala się ostatecznie, ale z błędem "coords is undefined", który ani w tekście ani w stacku nie zawiera żadnych informacji wskazujących na faktyczną przyczynę problemu (w naszym przykładzie: niezdefiniowany navigator.geolocation).

Reply:
Faktycznie, w trakcie pisania zmieniłem dość znacznie koncept i zapomniałem to zmienić, dzięki za uwagę